app/javascript/chat/components/ChannelImage.jsx

```diff
@@ -2,9 +2,26 @@ import { h } from 'preact';
 import PropTypes from 'prop-types';
 import { defaultChannelPropTypes } from '../../common-prop-types/channel-list-prop-type';
 
-export function ChannelImage(props) {
-  const { channel, newMessagesIndicator, discoverableChannel } = props;
+/**
+ * Returns an image to help users identify chat channels
+ *
+ * @param {object} channel - Contains information about the channel this image represents
+ * @param {string} newMessagesIndicator
+ * @param {boolean} discoverableChannel
+ *
+ * @example
+ * <ChannelImage
+     channel={channel}
+     newMessagesIndicator={newMessagesIndicator}
+     discoverableChannel={discoverableChannel}
+   />
```

Comment:
(suggestion / not blocking) I think for us to get the most out of JSDoc comments, we should try to provide the value types where we can instead of variable names, e.g. 

```suggestion
 * @example
 * <ChannelImage
     channel={{channel_name: "Example channel"}}
     newMessagesIndicator="new"
     discoverableChannel={true}
   />
```

Just to give people a bit more context beyond the prop name. It's not always useful, e.g. in the case of function prop types, but definitely for things like strings and boolean values